prov/rxm/src/rxm_cq.c

```diff
@@ -483,6 +524,7 @@ static int rxm_handle_remote_write(struct rxm_ep *rxm_ep,
 				"Unable to write remote write completion\n");
 		return ret;
 	}
+	rxm_cntr_report_comp(comp->flags, rxm_ep);
```

Comment:
For example, I prefer to see something couple of helpers (typing quickly, so names and structures need help):
```
static inline void util_inc_cntr(util_cntr *cntr)
{
    if (cntr)
        ofi_atomic_inc64(&cntr->val);
}

static inline void util_inc_cntrerr(util_cntr *cntr)
{
    if (cntr)
        ofi_atomic_inc64(&cntr->errval);
}
```
Then directly update the counter as needed:
```
util_inc_cntr(ep->remote_wr_cntr);
```
Note: I don't know if we can get away with just calling atomic inc, or if we need to go through the API call in order to ensure that any potential waiting threads are signaled.  The less overhead, the better.

Reply:
Unfortunately, this doesn't work if user use waiting threads. The add/adderr/set/seterr APIs signal about changing value